### DISCUSSION TITLE
Replace whole array data in useFieldArray

### DISCUSSION THREAD

**Discussion by darthmat** [react-hook-form/react-hook-form#1877] [Q&A] [answered]
Hi

Im using useFieldArray with:
`    const { fields, append, remove } = useFieldArray({
        control,
        name: 'aggregations',
    });`

that is filled with some data for example 10 rows. Is there a posibility to replace whole data at once with something like `setValue`? What i understand this `setValue` works only with inputs so something like that will not work:
`setValue('aggregations', newArray);`

**Comment by bluebill1049** [comment, marked as answer]
try it with `reset`

https://codesandbox.io/s/react-hook-form-usefieldarray-template-mxmuw
